propagation.go

```diff
@@ -272,7 +283,7 @@ func (p *binaryPropagator) Extract(abstractCarrier interface{}) (SpanContext, er
 // is converted to map[string]string { "key1" : "value1",
 //                                     "key2" : "value2",
 //                                     "key3" : "value3" }
-func parseCommaSeparatedMap(value string) map[string]string {
+func parseCommaSeparatedMap(value, headerKey string) map[string]string {
```

Comment:
this function and below - make them methods on the struct and read headers from the struct